db/sqldb/initiative.go

```diff
@@ -0,0 +1,217 @@
+package sqldb
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/RMI/pacta/db"
+	"github.com/RMI/pacta/pacta"
+	"github.com/jackc/pgx/v4"
+)
+
+const (
+	initiativeSelectColumns = `id, name, affiliation, public_description, internal_description,
+		requires_invitation_to_join, is_accepting_new_members, is_accepting_new_portfolios,
+		pacta_version_id, language, created_atdigest, created_at, COALESCE(is_default, false)`
+)
+
+func (d *DB) Initiative(tx db.Tx, id pacta.InitiativeID) (*pacta.Initiative, error) {
+	rows, err := d.query(tx, `
+		SELECT `+initiativeSelectColumns+`
+		FROM initiative 
+		WHERE id = $1;`, id)
+	if err != nil {
+		return nil, fmt.Errorf("querying initiative: %w", err)
+	}
+	pvs, err := rowsToInitiatives(rows)
+	if err != nil {
+		return nil, fmt.Errorf("translating rows to initiatives: %w", err)
+	}
+	return exactlyOne("initiative", id, pvs)
+}
+
+func (d *DB) Initiatives(tx db.Tx, ids []pacta.InitiativeID) (map[pacta.InitiativeID]*pacta.Initiative, error) {
```

Comment:
The reason I thought we might want to pursue this approach is that I looked through our multi-entity lookups in ADSCI, and almost every case where we call one of them (blobs, users) ends up converting the result to a map afterwards. 

Generally, I feel like the ergonomics of the array-return are unnatural for a situation where we pass in an array of (ideally the same size) IDs, since a novice user could easily assume the orders are the same in the two arrays, etc.

Reply:
I do think we're going to want to walk this back and have `map` be the special case and not the default, but you can implement few server endpoints and see how you feel about it.

Also, "multi-entity lookups" just sounds like we're doing `JOIN`s at our server layer instead of the database. Depending on the scenario, maybe that's fine, but in other cases we should just do it in the DB

Reply:
I look forward to seeing if you're right :)